packages/fetch-api/src/request.ts

```diff
@@ -5,40 +5,50 @@ export const DEFAULT_ERROR = 'Error fetching'
 const serverUrl = 'https://api.giphy.com/v1/'
 const identity = (i: any) => i
 const requestMap: { [key: string]: Promise<Result> } = {}
+
+class FetchError extends Error {
+    statusText: string
+    status: number
+    constructor(message: string, status: number = 0, statusText: string = '') {
+        super(message) // (1)
+        this.status = status
+        this.statusText = statusText
+    }
+}
 function request(
     url: string,
     normalizer: (a: any, pingbackType?: PingbackEventType) => any = identity,
     pingbackType?: PingbackEventType,
 ) {
     if (!requestMap[url]) {
-        requestMap[url] = new Promise<Result>(async (resolve, reject) => {
```

Comment:
This is an anit-pattern: https://stackoverflow.com/questions/43036229/is-it-an-anti-pattern-to-use-async-await-inside-of-a-new-promise-constructor

Was hard to refactor though!